File: examples/example_module.js
/**
 * Example JavaScript module to demonstrate README-Sync.
 * 
 * This module contains sample functions that will be parsed.
 */

/**
 * Calculate factorial of a number
 * @param {number} n - The number to calculate factorial for
 * @returns {number} The factorial of n
 */
export function factorial(n) {
  if (n <= 1) return 1;
  return n * factorial(n - 1);
}

/**
 * Fetch data from an API endpoint
 * @param {string} endpoint - API endpoint URL
 * @param {Object} options - Request options
 * @returns {Promise<Object>} Response data
 */
export async function fetchData(endpoint, options = {}) {
  const response = await fetch(endpoint, options);
  return response.json();
}

/**
 * User management class
 */
export class UserManager {
  constructor(apiUrl) {
    this.apiUrl = apiUrl;
    this.users = new Map();
  }

  /**
   * Get user by ID
   * @param {string} userId - User identifier
   * @returns {Promise<Object>} User object
   */
  async getUser(userId) {
    if (this.users.has(userId)) {
      return this.users.get(userId);
    }
    
    const user = await fetchData(`${this.apiUrl}/users/${userId}`);
    this.users.set(userId, user);
    return user;
  }

  /**
   * Create a new user
   * @param {Object} userData - User data
   * @returns {Promise<Object>} Created user
   */
  async createUser(userData) {
    const user = await fetchData(`${this.apiUrl}/users`, {
      method: 'POST',
      body: JSON.stringify(userData),
    });
    this.users.set(user.id, user);
    return user;
  }
}

/**
 * Utility function to format dates
 * @param {Date} date - Date to format
 * @param {string} format - Format string
 * @returns {string} Formatted date
 */
export const formatDate = (date, format = 'YYYY-MM-DD') => {
  // Implementation here
  return date.toISOString().split('T')[0];
};

File: examples/example_module.py
"""
Example Python module to demonstrate README-Sync.

This module contains sample functions that will be parsed by the AST parser.
"""


def calculate_fibonacci(n: int) -> int:
    """
    Calculate the nth Fibonacci number.
    
    Args:
        n: Position in the Fibonacci sequence (0-indexed)
        
    Returns:
        The nth Fibonacci number
        
    Raises:
        ValueError: If n is negative
    """
    if n < 0:
        raise ValueError("n must be non-negative")
    
    if n <= 1:
        return n
    
    return calculate_fibonacci(n - 1) + calculate_fibonacci(n - 2)


async def fetch_user_data(user_id: str, include_posts: bool = False) -> dict:
    """
    Fetch user data from the API.
    
    Args:
        user_id: Unique identifier for the user
        include_posts: Whether to include user's posts in the response
        
    Returns:
        Dictionary containing user data
    """
    # Simulated async operation
    return {
        "id": user_id,
        "name": "Example User",
        "posts": [] if include_posts else None
    }


class DataProcessor:
    """
    Process and transform data for analysis.
    
    This class provides methods for cleaning, transforming, and
    validating data before analysis.
    """
    
    def __init__(self, config: dict):
        """
        Initialize the data processor.
        
        Args:
            config: Configuration dictionary
        """
        self.config = config
        self.processed_count = 0
    
    def clean_data(self, data: list) -> list:
        """
        Remove invalid entries from data.
        
        Args:
            data: List of data entries
            
        Returns:
            Cleaned data list
        """
        cleaned = [item for item in data if self._is_valid(item)]
        self.processed_count += len(cleaned)
        return cleaned
    
    def _is_valid(self, item: dict) -> bool:
        """Check if a data item is valid."""
        return item is not None and len(item) > 0
    
    @staticmethod
    def transform(data: list, transformation: str) -> list:
        """
        Apply a transformation to the data.
        
        Args:
            data: Input data
            transformation: Type of transformation to apply
            
        Returns:
            Transformed data
        """
        if transformation == "uppercase":
            return [str(item).upper() for item in data]
        return data
